src/pages/AdvertisementForm.tsx

```diff
@@ -0,0 +1,223 @@
+                      })
+                    }}
+                    options={tokens.map((token) => ({
+                      label: token.symbol,
+                      value: token.symbol
+                    }))}
+                    onChange={(token) => {
+                      if (!token) return
+                      onChange(token.value)
+                    }}
+                    onBlur={onBlur}
+                  />
+                )}
+                rules={{ required: 'This field is required' }}
+              />
+
+              <ErrorMessage>{errors.currency2?.message || ''}</ErrorMessage>
+            </FormControl>
+
+            <FormControl>
+              <Label>Amount</Label>
+              <Input
+                {...register('amount', { required: 'This field is required' })}
+                placeholder="0.0"
+                error={!!errors.amount}
+              />
+              <ErrorMessage>{errors.amount?.message || ''}</ErrorMessage>
+            </FormControl>
+
+            {!active ? (
+              <ConnectWalletButton />
+            ) : !isValid ? (
+              <SubmitButton disabled>Fill the form</SubmitButton>
+            ) : submitting ? (
+              <SubmitButton disabled>Submitting...</SubmitButton>
+            ) : (
+              <SubmitButton>Create Advertisement</SubmitButton>
+            )}
+          </form>
+        </FormContainer>
+      </Body>
+    </Container>
+  )
+}
+
+const Container = styled.div`
+  width: 100vw;
+  display: flex;
+  justify-content: center;
+`
+
+const Body = styled.div`
+  width: 100%;
+  max-width: 600px;
+  padding: ${SPACE.XL} ${SPACE.XXL};
+`
+
+const PageHead = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+`
+
+const HeadLink = styled.a`
+  cursor: pointer;
+  font-weight: 600;
+`
+
+const FormContainer = styled.div`
+  background-color: ${({ theme }) => theme.surface};
+  color: ${({ theme }) => theme.onSurface};
+  box-shadow: 0 1px 4px ${({ theme }) => theme.shadow};
+  border-radius: ${RADIUS.M};
+  padding: ${SPACE.M};
+`
+
+const FormControl = styled.div`
+  width: 100%;
+  margin: ${SPACE.M} 0px;
+
+  &:first-child {
+    margin-top: 0;
+  }
+`
+
+const Input = styled.input<{ error?: boolean }>`
+  font-size: ${FONT_SIZE.L};
+  padding: ${SPACE.XS} ${SPACE.S};
+  border-radius: ${RADIUS.M};
+  border: solid 1px ${({ theme }) => theme.border};
+  ${({ error, theme }) => (error ? `border-color: ${theme.error}` : '')};
+  width: 100%;
+`
+
+const Label = styled.label`
+  display: block;
+  font-size: ${FONT_SIZE.S};
+  font-weight: 600;
+  color: ${({ theme }) => theme.onBackground};
+  margin-bottom: ${SPACE.XS};
+`
+
+const ErrorMessage = styled.div`
+  color: ${({ theme }) => theme.error};
+  font-size: ${FONT_SIZE.S};
+  height: 12px;
+`
+
+const SubmitButton = styled(PrimaryButton)`
+  font-size: ${FONT_SIZE.M};
+  font-weight: 600;
+  width: 100%;
+  padding: 12px;
+`
```

Comment:
How about moving "styled" definitions above the component declaration?

Reply:
@wanseob 
I think it's fine to put them after component declaration as long as the coding guideline is consistent in the project.
IMO, declaring component before styles is easier to read. But if you think putting styles before component declaration is better, I can put them at the top.
Or, it might be better to put them in another file. This, I see often in larger projects.
What do you think?

Reply:
Hey @tkmct yeah if you think that declaring first is easier to read I'm okay with that ;) I just thought that common convention was putting styles at the top. But I feel it'll be good to keep the styles in a single file until we have a well defined design system :)